builtin/providers/aws/resource_aws_emr_instance_group.go

```diff
@@ -0,0 +1,219 @@
+
+	// Guard against the chance of fetchEMRInstanceGroup returning nil group but
+	// not a emrInstanceGroupNotFound error
+	if group == nil {
+		log.Printf("[DEBUG] EMR Instance Group (%s) not found, removing", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", group.Name)
+	d.Set("instance_count", group.RequestedInstanceCount)
+	d.Set("running_instance_count", group.RunningInstanceCount)
+	d.Set("instance_type", group.InstanceType)
+	if group.Status != nil && group.Status.State != nil {
+		d.Set("status", group.Status.State)
+	}
+
+	return nil
+}
+
+func fetchAllEMRInstanceGroups(meta interface{}, clusterId string) ([]*emr.InstanceGroup, error) {
+	conn := meta.(*AWSClient).emrconn
+	req := &emr.ListInstanceGroupsInput{
+		ClusterId: aws.String(clusterId),
+	}
+
+	var groups []*emr.InstanceGroup
+	marker := aws.String("intitial")
+	for marker != nil {
+		log.Printf("[DEBUG] EMR Cluster Instance Marker: %s", *marker)
+		respGrps, errGrps := conn.ListInstanceGroups(req)
+		if errGrps != nil {
+			return nil, fmt.Errorf("[ERR] Error reading EMR cluster (%s): %s", clusterId, errGrps)
+		}
+		if respGrps == nil {
+			return nil, fmt.Errorf("[ERR] Error reading EMR Instance Group for cluster (%s)", clusterId)
+		}
+
+		if respGrps.InstanceGroups != nil {
+			for _, g := range respGrps.InstanceGroups {
+				groups = append(groups, g)
+			}
+		} else {
+			log.Printf("[DEBUG] EMR Instance Group list was empty")
+		}
+		marker = respGrps.Marker
+	}
+
+	if len(groups) == 0 {
+		return nil, fmt.Errorf("[WARN] No instance groups found for EMR Cluster (%s)", clusterId)
+	}
+
+	return groups, nil
+}
+
+func fetchEMRInstanceGroup(meta interface{}, clusterId, groupId string) (*emr.InstanceGroup, error) {
+	groups, err := fetchAllEMRInstanceGroups(meta, clusterId)
+	if err != nil {
+		return nil, err
+	}
+
+	var group *emr.InstanceGroup
+	for _, ig := range groups {
+		if groupId == *ig.Id {
+			group = ig
+			break
+		}
+	}
+
+	if group != nil {
+		return group, nil
+	}
+
+	return nil, emrInstanceGroupNotFound
+}
+
+func resourceAwsEMRInstanceGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+
+	log.Printf("[DEBUG] Modify EMR task group")
+	instanceCount := d.Get("instance_count").(int)
+
+	if d.HasChange("name") {
+		return fmt.Errorf("[WARN] Error updating task group, change name is not supported by api")
+	}
+
+	params := &emr.ModifyInstanceGroupsInput{
+		InstanceGroups: []*emr.InstanceGroupModifyConfig{
+			{
+				InstanceGroupId: aws.String(d.Id()),
+				InstanceCount:   aws.Int64(int64(instanceCount)),
+			},
+		},
+	}
+	resp, err := conn.ModifyInstanceGroups(params)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Modify EMR task group finished: %s", resp)
+
+	return nil
+}
+
+func resourceAwsEMRInstanceGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[WARN] AWS EMR Instance Group does not support DELETE; resizing cluster to zero before removing from state")
```

Comment:
Good addition to the logs here! One question, if this is at 0, does this still incur a cost?